narwhal/primary/src/proposer.rs

```diff
@@ -314,35 +311,25 @@ impl Proposer {
     }
 
     fn max_delay(&self) -> Duration {
-        match self.network_model {
-            // In partial synchrony, if this node is going to be the leader of the next
-            // round, we set a lower max timeout value to increase its chance of committing
-            // the leader.
-            NetworkModel::PartiallySynchronous
-                if self.committee.leader(self.round + 1).id() == self.authority_id =>
-            {
-                self.max_header_delay / 2
-            }
-
-            // Otherwise we keep the default timeout value.
-            _ => self.max_header_delay,
+        // If this node is going to be the leader of the next round, we set a lower max
+        // timeout value to increase its chance of being included in the dag.
+        if self.committee.leader(self.round + 1).id() == self.authority_id {
+            self.max_header_delay / 2
+        } else {
+            self.max_header_delay
         }
     }
 
     fn min_delay(&self) -> Duration {
-        match self.network_model {
-            // In partial synchrony, if this node is going to be the leader of the next
-            // round and there are more than 1 primary in the committee, we use a lower
-            // min delay value to increase the chance of committing the leader.
-            NetworkModel::PartiallySynchronous
-                if self.committee.size() > 1
-                    && self.committee.leader(self.round + 1).id() == self.authority_id =>
-            {
-                Duration::ZERO
-            }
-
-            // Otherwise we keep the default timeout value.
-            _ => self.min_header_delay,
+        // If this node is going to be the leader of the next round and there are more than
+        // 1 primary in the committee, we use a lower min delay value to increase the chance
+        // of committing the leader.
+        if self.committee.size() > 1
```

Comment:
Looking on this I am curious pragmatically which are the cases where `self.committee.size() < 1`  or at least anything where size < 4?

Reply:
`committee.size() == 1` is actually supported, so we need to give it a min header delay.

Reply:
we do? I mean do we ever expect to have a deployment of a single node?

Reply:
I didn't modify that, I simply kept the comment

Reply:
The background is that there are test(s) using `committee.size() == 1`, so I had to add this condition to make sure proposer does not monopolize the only thread available in simtests. IIRC I also added a yield in proposer, and this condition could be unnecessary for tests now. In future I'm not sure if we want to use `committee.size() == 1` more, e.g. for benchmarking, or reject this configuration.